token/js/src/extensions/tokenMetadata/state.ts

```diff
@@ -0,0 +1,105 @@
+import type { Commitment, Connection } from '@solana/web3.js';
+import { PublicKey } from '@solana/web3.js';
+import type { TokenMetadata } from '@solana/spl-token-metadata';
+import { Field, unpack } from '@solana/spl-token-metadata';
+
+import { TOKEN_2022_PROGRAM_ID } from '../../constants.js';
+import { ExtensionType, getExtensionData } from '../extensionType.js';
+import { getMint } from '../../state/mint.js';
+
+export const getNormalizedTokenMetadataField = (field: Field | string): string => {
+    if (field === Field.Name || field === 'Name' || field === 'name') {
+        return 'name';
+    }
+
+    if (field === Field.Symbol || field === 'Symbol' || field === 'symbol') {
+        return 'symbol';
+    }
+
+    if (field === Field.Uri || field === 'Uri' || field === 'uri') {
+        return 'uri';
+    }
+
+    return field;
+};
+
+export const updateTokenMetadata = (
```

Comment:
Any ideas how to simplify this function. It feels a bit overkill cause we only really need the change in length, but this lets us reuse the `pack` function.

Reply:
I removed support to update `mint` and `updateAuthority` - don't see why the js library would need it and it introduces some complexity